core/lib/src/response/named_file.rs

```diff
@@ -78,6 +78,31 @@ impl NamedFile {
     pub fn path(&self) -> &Path {
         self.0.as_path()
     }
+
+    /// Sets the default content type for the file
+    ///
+    /// If the file does not have an extension, the content type will
+    /// be set to this value. If no default is set and the file does not
+    /// have an extension, no content-type header will be sent.
```

Comment:
Perhaps:

Ordinarily, when responding, if the file path to `self` does not have an extension, or if the extension does not have a known Content-Type (see [`ContentType::from_extension()`]), no Content-Type is set in the response. By setting a default, the Content-Type of the response will be set to `content_type`.